pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -97,7 +101,10 @@ public CompletableFuture<Void> deleteTopicPoliciesAsync(TopicName topicName) {
 
     @Override
     public CompletableFuture<Void> updateTopicPoliciesAsync(TopicName topicName, TopicPolicies policies) {
-        return sendTopicPolicyEvent(topicName, ActionType.UPDATE, policies);
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        results.computeIfAbsent(topicName, k -> new LinkedBlockingDeque<>()).add(result);
```

Comment:
We should also remove the `topicName` from the `reulsts` if topic has been deleted

Reply:
fixed.